sky/shell/platform/ios/accessibility_bridge.h

```diff
@@ -0,0 +1,177 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SKY_SHELL_PLATFORM_IOS_SEMANTICS_LISTENER_IMPL_H_
+#define SKY_SHELL_PLATFORM_IOS_SEMANTICS_LISTENER_IMPL_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/macros.h"
+#include "base/memory/ref_counted.h"
+#include "base/memory/weak_ptr.h"
+#include "mojo/public/cpp/bindings/array.h"
+#include "mojo/public/cpp/bindings/strong_binding.h"
+#include "sky/engine/platform/geometry/FloatRect.h"
+#include "sky/services/semantics/semantics.mojom.h"
+#include "sky/shell/platform/ios/FlutterView.h"
+#include "third_party/skia/include/core/SkRect.h"
+#include "third_party/skia/include/utils/SkMatrix44.h"
+
+namespace sky {
+namespace shell {
+namespace a11y {
+
+class Node;
+
+typedef scoped_refptr<Node> NodePtr;
+
+// Class that listens for updates to the semantic tree from Dart code and issues
+// corresponding updates to the FlutterView's accessibility elements.
+//
+// The bridge is owned by the FlutterView that created it. It maintains a raw
+// pointer back to the view to enable bidirectional communication with the view
+// without introducing a circular reference. Since the strong binding herein may
+// destroy the bridge, the FlutterView maintains its ownership via a weak
+// pointer reference.
+class AccessibilityBridge final : public semantics::SemanticsListener {
+ public:
+  AccessibilityBridge(FlutterView*, semantics::SemanticsServer*);
+  ~AccessibilityBridge() override;
+
+  void UpdateSemanticsTree(mojo::Array<semantics::SemanticsNodePtr>) override;
+
+  base::WeakPtr<AccessibilityBridge> AsWeakPtr();
+
+ private:
+  typedef std::map<long, NodePtr> NodeMap;
+  friend class Node;
+
+  NodePtr UpdateNode(const semantics::SemanticsNodePtr& node);
+  void RemoveNode(NodePtr node);
+
+  NSArray* CreateAccessibleElements() const NS_RETURNS_RETAINED;
+
+  // See class docs above about ownership relationship
+  FlutterView* view_;
+  NodeMap nodes_;
+
+  mojo::StrongBinding<semantics::SemanticsListener> binding_;
+
+  base::WeakPtrFactory<AccessibilityBridge> weak_factory_;
+
+  DISALLOW_COPY_AND_ASSIGN(AccessibilityBridge);
+};
+
+struct Flags {
```

Comment:
I'd move these into the `.mm` file and put them in an anonymous namespace.  They don't seem to be used outside this file.